0.preprocess-sites/7.visualize-cell-summary.py

```diff
@@ -0,0 +1,249 @@
+foci_site_col = cell_args["foci_site_col"]
+
+output_resultsdir = summ_cell_args["output_resultsdir"]
+output_figuresdir = summ_cell_args["output_figuresdir"]
+cell_category_order = summ_cell_args["cell_category_order"]
+cell_category_colors = summ_cell_args["cell_category_colors"]
+
+cell_quality = CellQuality(quality_func)
+cell_category_dict = cell_quality.define_cell_quality()
+empty_cell_category = len(cell_category_dict) + 1
+cell_category_dict[empty_cell_category] = "Empty"
+cell_category_df = pd.DataFrame(cell_category_dict, index=["Cell_Class"])
+cell_category_list = list(cell_category_dict.values())
+
+# Read and Merge Data
+cell_quality_list = []
+metadata_list = []
+metadata_col_list = []
+metadata_col_list.append("Cell_Class")
+metadata_col_list.extend(cell_cols)
+metadata_col_list.extend(barcode_cols)
+metadata_col_list.extend(gene_cols)
+metadata_col_list.extend(spot_score_count_cols)
+metadata_col_list.extend(spot_score_mean_cols)
+metadata_col_list.append(cell_quality_col)
+metadata_col_list.append(foci_site_col)
+
+input_dir = pathlib.Path(input_basedir, batch, "paint")
+sites = os.listdir(input_dir)
+print(f"There are {len(sites)} sites.")
+
+for site in sites:
+
+    cell_count_file = pathlib.Path(input_dir, site, f"cell_counts_{site}.tsv")
+    metadata_file = pathlib.Path(input_dir, site, f"metadata_{site}.tsv.gz")
+
+    # Aggregates cell quality by site into single list
+    cell_quality_list.append(pd.read_csv(cell_count_file, sep='\t'))
+
+    # Aggregates metadata by site into a single list
+    metadata_df = (
+        pd
+        .read_csv(metadata_file, sep='\t')
+        .loc[:,metadata_col_list]
+        .reset_index(drop=True)
+    )
+
+    metadata_list.append(metadata_df)
+
+# Creates dataframe from cell quality list
+cell_count_df = pd.concat(cell_quality_list, axis="rows").reset_index(drop=True)
+
+# Assigns the Cell_Quality column to the category datatype and sets categories
+cell_count_df.loc[:, 'Cell_Quality'] = (
+    pd.Categorical(
+        cell_count_df.Cell_Quality,
+        categories=cell_category_order
+    )
+)
+
+# Assigns the Site column to the category datatype
+cell_count_df.loc[:, "site"] = (
+    pd.Categorical(
+        cell_count_df.site,
+        categories=(
+            cell_count_df
+            .groupby("site")
+            ["cell_count"]
+            .sum()
+            .sort_values(ascending=False)
+            .index
+            .tolist()
+        )
+    )
+)
+
+cell_count_df = cell_count_df.assign(
+    Plate=[x[0] for x in cell_count_df.site.str.split("-")],
+    Well=[x[1] for x in cell_count_df.site.str.split("-")],
+    Site=[x[2] for x in cell_count_df.site.str.split("-")]
+)
+
+output_file = pathlib.Path(output_resultsdir, "cells", "cell_count.tsv")
+#cell_count_df.to_csv(output_file, sep='\t', index=False)
+
+print(cell_count_df.shape)
+cell_count_df.head(10)
+
+# Graph: Cell count with all wells in same graph
```

Comment:
This is currently plotting with sites in numerical order and not sorted by cell_sum order, but should plot in the correct order once the previous .groupby issue is fixed.